### PR TITLE
Introduced model.Model#convertToDeltas

### LINKED ISSUE #1380: Introduce model.Model#convertToDeltas
Introduce a function that will take a `model.DocumentFragment` instance and convert it to an array of deltas which, when applied, will re-create the document fragment. Should return one `InsertDelta` with all the nodes and zero or more `MarkerDelta`s to re-create all the markers that were set on the `model.DocumentFragment`.

This is required by Letters and also should be useful for any compression algorithms.

### PR DESCRIPTION
**Pull request by scofalik**
### Suggested merge commit message ([convention](https://github.com/ckeditor/ckeditor5-design/wiki/Git-commit-message-convention))

Feature: Introduced `model.Model#convertToDeltas`. Closes #1380.

### CODE REVIEW

File: src/model/model.js [hunk 1]
```diff
@@ -382,6 +388,48 @@ export default class Model {
...
+	 * inserts all the nodes and zero or more {@link module:engine/model/operation/markeroperation~MarkerOperation}, depending
+	 * on {@link module:engine/model/documentfragment~DocumentFragment#markers markers set on the document fragment}.
+	 *
+	 * @param {module:engine/model/documentfragment~DocumentFragment} documentFragment Document fragment to be converted to deltas.
+	 * @param {String} [rootName] Name of a root element which will be base for the operations. The returned insert
+	 * operation will insert the nodes inside the given root. If not set, the main root will be used.
+	 * @returns {Array.<module:engine/model/operation/operation~Operation>} Operations which, when applied, will recreate given
+	 * `documentFragment` inside the specified root.
+	 */
+	convertToDeltas( documentFragment, rootName ) {
```
Comment (Reinmar) [inline src/model/model.js]: If you can think of some other name, which avoids the word "convert", I'm sure @pjasiun would be happy :P

Comment (pjasiun) [reply src/model/model.js]: Yep. And `transformToDeltas` won't be good either.

File: src/model/model.js [hunk 2]
```diff
@@ -382,6 +388,48 @@ export default class Model {
...
+	/**
+	 * Creates deltas from given {@link module:engine/model/documentfragment~DocumentFragment}.
+	 *
+	 * Returned array contains exactly one {@link module:engine/model/operation/insertoperation~InsertOperation} which
+	 * inserts all the nodes and zero or more {@link module:engine/model/operation/markeroperation~MarkerOperation}, depending
+	 * on {@link module:engine/model/documentfragment~DocumentFragment#markers markers set on the document fragment}.
+	 *
+	 * @param {module:engine/model/documentfragment~DocumentFragment} documentFragment Document fragment to be converted to deltas.
+	 * @param {String} [rootName] Name of a root element which will be base for the operations. The returned insert
+	 * operation will insert the nodes inside the given root. If not set, the main root will be used.
```
Comment (pjasiun) [inline src/model/model.js]: Can't you just create the tree in the `RootElement`?

File: src/model/model.js (hunk 1 above)
Comment (pjasiun) [reply src/model/model.js]: In the F2F talk, we agreed it will have no parameters and will serialize model to deltas. It can be called `getAsDeltas`

**Comment by scofalik** [comment]
Eventually, we agreed with PJ that we can create a `model.Model` instance and rename the method to `getAsDeltas()`. But then we concluded that we could just take the deltas straight from the history. So this method is not needed for now.

**Comment by Reinmar** [comment]
My pleasure :D

**Pull request closed without merging**